### PR TITLE
More work on settings editor

### PR DESCRIPTION
**Pull request by 3nids**
* move QgsSettingsEditorWidgetWrapper to core
* allow to specify a specific editor for a setting
* fix crash with use of PyQgsSettingsEntryEnumFlag in settings tree widget 
* Python implementation of enum settings editor widget wrapper (https://github.com/qgis/QGIS/pull/59006/commits/df21a64a5952bc0961d08acc98ceec338df1e2bf)

### CODE REVIEW

File: src/core/settings/qgssettingsentry.h [hunk 1]
```diff
@@ -352,6 +353,22 @@ class CORE_EXPORT QgsSettingsEntryBase
      */
     bool hasChanged() const { return mHasChanged; }
 
+    /**
+     * Sets an editor widget wrapper for the setting
+     * Normally, editor widgets are automatically determined and build using QgsSettingsEditorWidgetRegistry.
```
Comment (rouault) [inline src/core/settings/qgssettingsentry.h:358]:
```suggestion
     * Normally, editor widgets are automatically determined and built using QgsSettingsEditorWidgetRegistry.
```

File: src/core/settings/qgssettingsentry.h [hunk 2]
```diff
@@ -352,6 +353,22 @@ class CORE_EXPORT QgsSettingsEntryBase
...
+     * This might be useful for plugins to define a custom implementation.
+     * \since QGIS 3.40
+     */
+    void setEditorWidgetWrapper( QgsSettingsEditorWidgetWrapper *editorWrapper SIP_TRANSFER ) {mEditorWrapper = editorWrapper; }
+
+    /**
+     * Returns the editor widget wrapper for the setting
+     * Normally, editor widgets are automatically determined and build using QgsSettingsEditorWidgetRegistry and
```
Comment (rouault) [inline src/core/settings/qgssettingsentry.h:366]:
```suggestion
     * Normally, editor widgets are automatically determined and built using QgsSettingsEditorWidgetRegistry and
```

**Comment by nyalldawson** [comment]
> move QgsSettingsEditorWidgetWrapper to core

I'm -1 to this -- it's definitely a GUI focused class, and is full of qwidget references. Why should it be in core?

**Comment by 3nids** [comment]
To be able to set an editor for a setting (and avoid creating a registry for this).

It follows the same logic than `QgsEditorWidgetWrapper`, all the implementations are in gui.

There is no GUI includes in the header, and only `QDialog` and `QWidget` in the source.

**Comment by nyalldawson** [comment]
@3nids 

That still screams of bad API to me. I personally think a registry in the GUI lib would be the correct approach

**Comment by 3nids** [comment]
OK, I'll have another look. In the mean time, I opened another PR with the still relevant stuff.

**Pull request closed without merging**